src/player/chat.js

```diff
@@ -48,30 +83,16 @@ export default {
         log.info({ sender: client.uuid, message }, 'Message')
       }
 
-      const formatted_message = message.split(/(%item\d%)/).map(part => {
-        if (part.match(/(%item\d%)/)) {
-          const slot_number = parseInt(part.match(/\d/)[0]) + 36 // For the player 0 is the first item in hotbar. But for the game the hotbat begin at 36.
-          const { inventory } = get_state()
-          const item = inventory[slot_number]
-          if (item) {
-            const { type, count } = item
-            return slot_to_chat(item_to_slot(items[type], count))
-          }
-        }
-        return { text: part }
-      })
-
       world_chat_msg({
         world,
-        message: {
-          translate: 'chat.type.text',
-          with: [
-            {
-              text: client.username,
-            },
-            formatted_message,
-          ],
-        },
+        message: [
+          { text: `${nickname}: `, ...Formats.BASE, italic: false },
+          ...message.split(' ').map(word => {
+            for (const pattern in chat_mapper)
+              if (word.match(pattern)) return chat_mapper[pattern](word)
+            return { text: `${word} `, ...Formats.BASE, italic: false }
+          }),
+        ],
```

Comment:
I would prefer for us to keep the translating key (cc @Sceat) as this allows to not hardcode the format here and use the one that we put in the ressource pack

```suggestion
        message: {
          translate: 'chat.type.text',
          with: [
            {
              text: client.username,
            },
            message.split(' ').map(word => {
              for (const pattern in chat_mapper)
                if (word.match(pattern)) return chat_mapper[pattern](word)
              return { text: `${word} `, ...Formats.BASE, italic: false }
            }),
          ],
        },
```

Reply:
oh this is resource handled ? interesting